packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/middleware/logging.ts

```diff
@@ -0,0 +1,40 @@
+import { NextFunction, Request, Response } from 'express'
+import { v4 as uuidv4 } from 'uuid'
+import { logger } from '../logger'
+
+export const incomingRequestLogger = (
+  request: Request,
+  response: Response,
+  next: NextFunction
+) => {
+  const startTime = new Date().getTime()
+  response.locals.requestStartTime = startTime
+  const requestId = uuidv4()
```

Comment:
nit: client does send a request id under the `X-Request-Id` header, you could capture that here
something like this
```
const requestId = request.headers["X-Request-Id"] || uuidv4()
```